projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -125,6 +127,15 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseCompone
     *      <igx-column field="Description"  [dataType]='string'></igx-column>
     * </igx-hierarchical-grid>
     * ```
+    *
+    * Two-way data binding.
+    * ```html
+    * <igx-hierarchical-grid [primaryKey]="'ID'" [data]="Data" [autoGenerate]="false" [(hierarchicalState)]="hgridState">
```

Comment:
@MayaKirova Setting two-way data binding for `hierarchicalState` throws 'Maximum call stack size exceeded' exception.